state/hashesCollector/dataTrieHashesCollector.go

```diff
@@ -0,0 +1,62 @@
+package hashesCollector
+
+import (
+	"sync"
+
+	"github.com/multiversx/mx-chain-go/common"
+)
+
+type dataTrieHashesCollector struct {
+	oldHashes common.ModifiedHashes
+	newHashes common.ModifiedHashes
+
+	sync.RWMutex
+}
+
+// NewDataTrieHashesCollector creates a new instance of dataTrieHashesCollector.
+// This collector is used to collect hashes related to the data trie.
+func NewDataTrieHashesCollector() *dataTrieHashesCollector {
```

Comment:
You make a new one for this every single time ? and that is why you do not have delete function ?

I think adding some capacity on the creation of the oldHashes / newHashes can make processing faster.